src/ChatPage.cpp

```diff
@@ -525,7 +525,7 @@ ChatPage::ChatPage(QSharedPointer<UserSettings> userSettings, QWidget *parent)
                                 hasNotifications = true;
                 }
 
-                if (hasNotifications && userSettings_->hasDesktopNotifications())
+                if (hasNotifications && userSettings_->hasNotifications())
```

Comment:
This means, that the window alert is only emitted, when `alertOnNotification` and `hasNotifications` is true. Is that intentional?

If it is, can you make the toggle for alerts depend on the hasNotifications toggle being in? If it isn't, I guess you want to change/move this check.

Reply:
`hasNotifications()` is an _or_, not and _and_.
There is also no extra notification setting for the tray icon. I didn't see any tray-related code for the desktop notifications - did I miss it?

Reply:
Woops, I didn't see this change: https://github.com/Nheko-Reborn/nheko/pull/218/files#diff-2d3e7e9f4969499877551aa3cfe5741bR115

With that it makes sense, that this is an or and the toggles don't need to depend on one another. The tray stuff was just for if the buttons depend on one another, you could have looked there on how it is done.